pkg/infoschema/test/infoschemav2test/v2_test.go

```diff
@@ -534,3 +535,26 @@ func TestSnapshotInfoschemaReader(t *testing.T) {
 	sql = fmt.Sprintf("select * from INFORMATION_SCHEMA.TABLES as of timestamp '%s' where table_schema = 'issue55827'", timeStr)
 	tk.MustQuery(sql).Check(testkit.Rows())
 }
+
+func TestInfoSchemaCachedAutoIncrement(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	autoid.SetStep(1)
+	tk.MustExec("set @@global.tidb_schema_cache_size = 0;")
+	tk.MustExec("create table t (a int primary key auto_increment);")
+	autoIncQuery := "select auto_increment from information_schema.tables where table_name = 't' and table_schema = 'test';"
+
+	tk.MustQuery(autoIncQuery).Check(testkit.Rows("0"))
+	tk.MustExec("insert into t values (),(),();")
+	tk.MustQuery(autoIncQuery).Check(testkit.Rows("4"))
+
+	tk.MustExec("set @@global.tidb_schema_cache_size = 1024 * 1024 * 1024;")
+	tk.MustExec("create table t1 (a int);") // trigger infoschema cache reload
+	tk.MustQuery(autoIncQuery).Check(testkit.Rows("0"))
+	tk.MustExec("insert into t values ();")
+	tk.MustQuery(autoIncQuery).Check(testkit.Rows("5"))
+	tk.MustExec("set @@global.tidb_schema_cache_size = 0;")
```

Comment:
There is an API you can use to evict table cache directly, although trigger reload can get the same effect.

```
	is := dom.InfoSchema()
	ok, v2 := infoschema.IsV2(is)
	require.True(t, ok)
	v2.EvictTable(model.NewCIStr("test"), model.NewCIStr("t1"))
```

Reply:
The schema is not v2 if we don't reload.